Windows compilation fails

Compiling an application using this library to a windows binary results in the following build errors

```
GOOS=windows GOARCH=amd64 go build
# github.com/rabbitmq/rabbitmq-stream-go-client/pkg/stream
vendor/github.com/rabbitmq/rabbitmq-stream-go-client/pkg/stream/client.go:97:38: cannot use int(fd) (type int) as type syscall.Handle in argument to syscall.SetsockoptInt
vendor/github.com/rabbitmq/rabbitmq-stream-go-client/pkg/stream/client.go:104:37: cannot use int(fd) (type int) as type syscall.Handle in argument to syscall.SetsockoptInt
```

--- a/pkg/stream/client.go
+++ b/pkg/stream/client.go
@@ -9,10 +9,8 @@ import (
 	"math/rand"
 	"net"
 	"net/url"
-	"runtime"
 	"sync"
 	"sync/atomic"
-	"syscall"
 	"time"
 )
 
@@ -106,24 +104,7 @@ func (c *Client) connect() error {
 		c.tuneState.requestedMaxFrameSize = 1048576
 
 		var dialer = &net.Dialer{
-			Control: func(network, address string, c syscall.RawConn) error {
-				return c.Control(func(fd uintptr) {
-					err := syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_RCVBUF, defaultSocketBuffer)
-					runtime.KeepAlive(fd)
-					if err != nil {
-						logs.LogError("Set socket option error: %s", err)
-						return
-					}
-
-					err = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_SNDBUF, defaultSocketBuffer)
-					runtime.KeepAlive(fd)
-					if err != nil {
-						logs.LogError("Set socket option error: %s", err)
-						return
-					}
-
-				})
-			},
+			Control: controlFunc,
 		}
 
 		var connection net.Conn

--- a/pkg/stream/dialer_posix.go
+++ b/pkg/stream/dialer_posix.go
@@ -1,0 +1,33 @@
+// +build !windows
+
+package stream
+
+import (
+	"runtime"
+	"syscall"
+
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/logs"
+)
+
+var controlFunc func(network, address string, c syscall.RawConn) error
+
+func init() {
+	controlFunc = func(network, address string, c syscall.RawConn) error {
+		return c.Control(func(fd uintptr) {
+			err := syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_RCVBUF, defaultSocketBuffer)
+			runtime.KeepAlive(fd)
+			if err != nil {
+				logs.LogError("Set socket option error: %s", err)
+				return
+			}
+
+			err = syscall.SetsockoptInt(int(fd), syscall.SOL_SOCKET, syscall.SO_SNDBUF, defaultSocketBuffer)
+			runtime.KeepAlive(fd)
+			if err != nil {
+				logs.LogError("Set socket option error: %s", err)
+				return
+			}
+
+		})
+	}
+}

--- a/pkg/stream/dialer_windows.go
+++ b/pkg/stream/dialer_windows.go
@@ -1,0 +1,33 @@
+// +build windows
+
+package stream
+
+import (
+	"runtime"
+	"syscall"
+
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/logs"
+)
+
+var controlFunc func(network, address string, c syscall.RawConn) error
+
+func init() {
+	controlFunc = func(network, address string, c syscall.RawConn) error {
+		return c.Control(func(fd uintptr) {
+			err := syscall.SetsockoptInt(syscall.Handle(fd), syscall.SOL_SOCKET, syscall.SO_RCVBUF, defaultSocketBuffer)
+			runtime.KeepAlive(fd)
+			if err != nil {
+				logs.LogError("Set socket option error: %s", err)
+				return
+			}
+
+			err = syscall.SetsockoptInt(syscall.Handle(fd), syscall.SOL_SOCKET, syscall.SO_SNDBUF, defaultSocketBuffer)
+			runtime.KeepAlive(fd)
+			if err != nil {
+				logs.LogError("Set socket option error: %s", err)
+				return
+			}
+
+		})
+	}
+}

